cpp/models/ode_secir/parameters.h

```diff
@@ -381,82 +385,97 @@ class Parameters : public ParametersBase
                             this->get<IncubationTime>()[i], 2.0);
                 this->get<IncubationTime>()[i] = 2.0;
+                corrected                      = 1;
             }
 
             if (2 * this->get<SerialInterval>()[i] < this->get<IncubationTime>()[i] + 1.0) {
                 log_warning("Constraint check: Parameter SerialInterval changed from {:.4f} to {:.4f}",
                             this->get<SerialInterval>()[i], 0.5 * this->get<IncubationTime>()[i] + 0.5);
                 this->get<SerialInterval>()[i] = 0.5 * this->get<IncubationTime>()[i] + 0.5;
+                corrected                      = 1;
             }
             else if (this->get<SerialInterval>()[i] > this->get<IncubationTime>()[i] - 0.5) {
                 log_warning("Constraint check: Parameter SerialInterval changed from {:.4f} to {:.4f}",
                             this->get<SerialInterval>()[i], this->get<IncubationTime>()[i] - 0.5);
                 this->get<SerialInterval>()[i] = this->get<IncubationTime>()[i] - 0.5;
+                corrected                      = 1;
             }
 
             if (this->get<TimeInfectedSymptoms>()[i] < 1.0) {
                 log_warning("Constraint check: Parameter TimeInfectedSymptoms changed from {:.4f} to {:.4f}",
                             this->get<TimeInfectedSymptoms>()[i], 1.0);
                 this->get<TimeInfectedSymptoms>()[i] = 1.0;
+                corrected                            = 1;
             }
 
             if (this->get<TimeInfectedSevere>()[i] < 1.0) {
                 log_warning("Constraint check: Parameter TimeInfectedSevere changed from {:.4f} to {:.4f}",
                             this->get<TimeInfectedSevere>()[i], 1.0);
                 this->get<TimeInfectedSevere>()[i] = 1.0;
+                corrected                          = 1;
             }
 
             if (this->get<TimeInfectedCritical>()[i] < 1.0) {
                 log_warning("Constraint check: Parameter TimeInfectedCritical changed from {:.4f} to {:.4f}",
                             this->get<TimeInfectedCritical>()[i], 1.0);
                 this->get<TimeInfectedCritical>()[i] = 1.0;
+                corrected                            = 1;
             }
 
-            if (this->get<TransmissionProbabilityOnContact>()[i] < 0.0) {
+            if (this->get<TransmissionProbabilityOnContact>()[i] < 0.0 ||
+                this->get<TransmissionProbabilityOnContact>()[i] > 1.0) {
                 log_warning(
                     "Constraint check: Parameter TransmissionProbabilityOnContact changed from {:0.4f} to {:d} ",
-                    this->get<TransmissionProbabilityOnContact>()[i], 0);
-                this->get<TransmissionProbabilityOnContact>()[i] = 0;
+                    this->get<TransmissionProbabilityOnContact>()[i], 0.5);
+                this->get<TransmissionProbabilityOnContact>()[i] = 0.5;
+                corrected                                        = 1;
```

Comment:
Why do you set the probability to 0.5 here whereas for the other parameters you set the value to the minimum possible?

Reply:
My intention was to take a halfway realistic value, since we have no infection at all for 0. But for a wrong input this seems to be a reasonable choice of the parameter and it is consistent to the choice of the other parameters by the apply_constraint function.